rust/src/mask/mod.rs

```diff
@@ -1,13 +1,204 @@
+//! let config = MaskConfig {
+//!     group_type: GroupType::Prime,
+//!     data_type: DataType::F32,
+//!     bound_type: BoundType::B0,
+//!     model_type: ModelType::M3,
+//! };
+//!
+//! // mask the local models
+//! let (local_mask_seed_1, masked_local_model_1) = Masker::new(config).mask(scalar, local_model_1);
+//! let (local_mask_seed_2, masked_local_model_2) = Masker::new(config).mask(scalar, local_model_2);
+//!
+//! // derive the masks of the local masked models
+//! let local_mask_1 = local_mask_seed_1.derive_mask(number_weights, config);
+//! let local_mask_2 = local_mask_seed_2.derive_mask(number_weights, config);
+//! ```
+//!
+//! ## Aggregation
+//! Masked models can be aggregated via an [`Aggregation`]. Masks can be aggregated via an
+//! [`Aggregation`] as well, since an aggregated masked model can only be unmasked by the
+//! aggregated masks belonging to the masked models. It should always be validated that the
+//! aggregation may be safely performed wrt the chosen masking configuration before actually
+//! aggregating to avoid the possible loss of information.
+//!
+//! ```
+//! # use xain_fl::mask::{Aggregation, BoundType, DataType, FromPrimitives, GroupType, MaskConfig, Masker, MaskObject, Model, ModelType};
+//! # let number_weights = 10;
+//! # let scalar = 0.5;
+//! # let local_model_1 = Model::from_primitives_bounded(vec![0_f32; number_weights].into_iter());
+//! # let local_model_2 = Model::from_primitives_bounded(vec![1_f32; number_weights].into_iter());
+//! # let config = MaskConfig { group_type: GroupType::Prime, data_type: DataType::F32, bound_type: BoundType::B0, model_type: ModelType::M3};
+//! # let (local_mask_seed_1, masked_local_model_1) = Masker::new(config).mask(scalar, local_model_1);
+//! # let (local_mask_seed_2, masked_local_model_2) = Masker::new(config).mask(scalar, local_model_2);
+//! # let local_mask_1 = local_mask_seed_1.derive_mask(number_weights, config);
+//! # let local_mask_2 = local_mask_seed_2.derive_mask(number_weights, config);
+//! // aggregate the local masks
+//! let mut mask_aggregator = Aggregation::new(config);
+//! if let Ok(_) = mask_aggregator.validate_aggregation(&local_mask_1) {
+//!     mask_aggregator.aggregate(local_mask_1);
+//! };
+//! if let Ok(_) = mask_aggregator.validate_aggregation(&local_mask_2) {
+//!     mask_aggregator.aggregate(local_mask_2);
+//! };
+//! let global_mask: MaskObject = mask_aggregator.into();
+//!
+//! // aggregate the local masked models
+//! let mut model_aggregator = Aggregation::new(config);
+//! if let Ok(_) = model_aggregator.validate_aggregation(&masked_local_model_1) {
+//!     model_aggregator.aggregate(masked_local_model_1);
+//! };
+//! if let Ok(_) = model_aggregator.validate_aggregation(&masked_local_model_2) {
+//!     model_aggregator.aggregate(masked_local_model_2);
+//! };
+//! ```
+//!
+//! ## Unmasking
+//! A masked model can be unmasked by the corresponding mask via an [`Aggregation`] in the end. It
```

Comment:
```suggestion
//! A masked model can be unmasked by the corresponding mask via an [`Aggregation`]. It
```
or alternatively, elaborate on "in the end"

Reply:
changed.